### PR TITLE
General clean up: openedx.yaml, pip-tools, pylint, etc.

### PR DESCRIPTION
**Pull request by kdmccormick**
This is a general tooling PR, in preparation for adding API documentation tooling to this repo.

It sets the owner to @nasthagiri , uses pip-tools for test requirements, fixes PEP-8 violations, switches to pycodestyle, starts running quality checks again, fixes a bunch of pylint errors, and installs isort.

Note that pylint Convention and Refactor messages are disabled because there are so many of them.

FYI @nedbat

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,8 +12,10 @@ env:
 matrix:
  include:
   - python: 3.6
-    env: TOXENV=quality,docs
-    python: 3.6
+    env: TOXENV=quality
+  - python: 3.6
+    env: TOXENV=docs
+  - python: 3.6
```
Comment (nedbat) [inline .travis.yml]: We have 2.7 and 3.5 above.

File: Makefile [hunk 2]
```diff
@@ -10,12 +10,37 @@ clean:
...
+	# `make upgrade` removes Django from test_requirements.txt,
+	# so we manually install it here.
+	pip install django~=1.11  
 	pip install -r docs/requirements.txt
 
 test:
 	tox
 
-.PHONY: clean, quality, requirements
+.PHONY: clean quality requirements _upgrade upgrade piptools
```
Comment (nedbat) [inline Makefile]: We could add "help" here also.

File: edx_rest_framework_extensions/auth/jwt/authentication.py [hunk 3]
```diff
@@ -83,11 +87,23 @@ def authenticate_credentials(self, payload):
...
                                 if key in current_value:
                                     if current_value[key] != value:
-                                        logger.info('Updating attribute %s[%s] for user %s with value %s', attr, key, user.id, value)
+                                        logger.info(
+                                            'Updating attribute %s[%s] for user %s with value %s',
+                                            attr,
+                                            key,
+                                            user.id,
+                                            value,
+                                        )
```
Comment (nedbat) [inline edx_rest_framework_extensions/auth/jwt/authentication.py]: Ugh, whatever.

File: edx_rest_framework_extensions/tests/test_middleware.py [hunk 4]
```diff
@@ -73,7 +73,12 @@ def test_request_standard_user_agent_metrics(self, mock_set_custom_metric):
     )
     @ddt.unpack
     @patch('edx_django_utils.monitoring.set_custom_metric')
-    def test_request_auth_type_token_metric(self, token, expected_token_type, mock_set_custom_metric):
+    def test_request_auth_type_token_metric(
+            self,
+            token,
+            expected_token_type,
+            mock_set_custom_metric,
+    ):
```
Comment (nedbat) [inline edx_rest_framework_extensions/tests/test_middleware.py]: Why split out this line?

File: edx_rest_framework_extensions/tests/test_middleware.py [hunk 5]
```diff
@@ -91,7 +96,9 @@ def test_request_auth_type_session_metric(self, mock_set_custom_metric):
         self.request.user = UserFactory()
 
         self.middleware.process_response(self.request, None)
-        mock_set_custom_metric.assert_any_call('request_auth_type', 'session-or-unknown')
+        mock_set_custom_metric.assert_any_call(
+            'request_auth_type', 'session-or-unknown'
+        )
```
Comment (nedbat) [inline edx_rest_framework_extensions/tests/test_middleware.py]: Why split this line?

File: .travis.yml (hunk 1 above)
Comment (kdmccormick) [reply .travis.yml]: What are you suggesting?

File: edx_rest_framework_extensions/tests/test_middleware.py (hunk 4 above)
Comment (kdmccormick) [reply edx_rest_framework_extensions/tests/test_middleware.py]: mistakenly thought it was over 120chars. same mistake below

File: edx_rest_framework_extensions/auth/jwt/authentication.py (hunk 3 above)
Comment (kdmccormick) [reply edx_rest_framework_extensions/auth/jwt/authentication.py]: Yeah...

File: .travis.yml (hunk 1 above)
Comment (kdmccormick) [reply .travis.yml]: I simplified this a bit; hopefully in line with what you were thinking.

**Comment by kdmccormick** [comment]
closed in favor of #94

**Pull request closed without merging**